pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -0,0 +1,84 @@
+package pipelinerun
+
+import (
+	"context"
+	"github.com/redhat-appstudio/jvm-build-service/pkg/reconciler/dependencybuild"
+	pipelinev1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/pkg/apis"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"time"
+
+	"github.com/redhat-appstudio/jvm-build-service/pkg/apis/jvmbuildservice/v1alpha1"
+)
+
+const (
+	//TODO eventually we'll need to decide if we want to make this tuneable
+	contextTimeout = 300 * time.Second
+)
+
+type ReconcilePipelineRunRequest struct {
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+func newReconciler(mgr ctrl.Manager) reconcile.Reconciler {
+	return &ReconcilePipelineRunRequest{
+		client: mgr.GetClient(),
+		scheme: mgr.GetScheme(),
+	}
+}
+
+func (r *ReconcilePipelineRunRequest) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	// Set the ctx to be Background, as the top-level context for incoming requests.
+	ctx, cancel := context.WithTimeout(ctx, contextTimeout)
+	defer cancel()
+	//log := log.FromContext(ctx)
+	pr := pipelinev1beta1.PipelineRun{}
+	err := r.client.Get(ctx, request.NamespacedName, &pr)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+	//if there is no label then ignore it
+	abrNameForLabel, existingLabel := pr.Labels[dependencybuild.IdLabel]
```

Comment:
so this is what we discussed last night, so to be clear, first pass this is what I was expecting

but in case either a) I failed to mention this tidbit, or b) it got lost in the torrent of details we went over, even if we continue down this "explicit" reconciler approach (vs. the "shipwright" approach I've been mentioning), we'll most likely want to employ the options available when setting up the controller in the first place so that it filters pipelineruns so we only get events for the ones with our label

The controller runtime doc that tries to describe this is at https://github.com/kubernetes-sigs/controller-runtime/blob/master/designs/use-selectors-at-cache.md

Then in our code, in the `SetupNewReconcilerWithManager` methods, we would switch from the "simpler" `ctrl.NewControllerManagedBy` call to the more "powerfull/flexible" `ctrl.NewManager` call, where we set the `Options.NewCache` function.

@stuartwdouglas ^^

Reply:
@gabemontero I have added a commit that attempts to add what you describe, although the code is a bit different to what you mentioned so I am not sure if it is right.

Reply:
ok got a few minutes taking a look now @stuartwdouglas